### PR TITLE
[foxy] port packages

### PR DESCRIPTION
**Pull request by Deleh**
Ports `cob_perception_msgs` to ROS 2.

includes: https://github.com/ipa320/cob_perception_common/pull/107

### CODE REVIEW

**Review by ipa-rmb — DISMISSED** [review]

File: cob_3d_mapping_msgs/cob_3d_mapping_msgs_mapping_rules.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  ros2_package_name: 'cob_3d_mapping_msgs'
+  ros2_message_name: 'SimilarityScore'
+  fields_1_to_2:
+    ID: 'id'
+    prob: 'prob'
+# add mapping rule for:
+#  - action/PlaneExtraction.action
+#  - action/TableObjectCluster.action
+#  - action/Trigger.action
+#  - srv/MoveToTable.srv
```
Comment (fmessmer) [inline cob_3d_mapping_msgs/cob_3d_mapping_msgs_mapping_rules.yaml]: LoyVanBeek please have a look
maybe you can help configure the missing mapping rules

**Review by fmessmer — CHANGES_REQUESTED** [review]

File: cob_3d_mapping_msgs/cob_3d_mapping_msgs_mapping_rules.yaml (hunk 1 above)
Comment (LoyVanBeek) [reply cob_3d_mapping_msgs/cob_3d_mapping_msgs_mapping_rules.yaml]: I tried mapping those in https://github.com/Deleh/cob_perception_common/pull/1 but doesn't compile. 
Not sure if that is my or `ros1_bridge`'s fault yet.

AFAIK, there is no docs on how to write mapping rules for Actions yet (service example is in https://github.com/mojin-robotics/ros1_bridge/blame/mojin-devel/doc/index.rst#L144), so it's 'Use the Source, Loy' :star2:  for that stuff for now. I looked at https://github.com/mojin-robotics/ros1_bridge/blob/mojin-devel/ros1_bridge/__init__.py#L622 for that.

File: cob_3d_mapping_msgs/CMakeLists.txt [hunk 2]
```diff
@@ -1,91 +1,77 @@
...
-    action
-  FILES
-    PlaneExtraction.action
-    TableObjectCluster.action
-    Trigger.action
+set(action_files
+  # "action/PlaneExtraction.action"
+  # "action/TableObjectCluster.action"
+  # "action/Trigger.action"
 )
```
Comment (fmessmer) [inline cob_3d_mapping_msgs/CMakeLists.txt:32]: commented due to issues with `ros1_bridge`

File: cob_3d_mapping_msgs/cob_3d_mapping_msgs_mapping_rules.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#   ros2_action_name: 'TableObjectCluster'
+#   feedback_fields_1_to_2:
+#     currentStep: 'current_step'
+# -
+#   ros1_package_name: 'cob_3d_mapping_msgs'
+#   ros1_action_name: 'Trigger'
+#   ros2_package_name: 'cob_3d_mapping_msgs'
+#   ros2_action_name: 'Trigger'
+#   feedback_fields_1_to_2:
+#     currentStep: 'current_step'
```
Comment (fmessmer) [inline cob_3d_mapping_msgs/cob_3d_mapping_msgs_mapping_rules.yaml:54]: commented due to issues with `ros1_bridge`

**Review by fmessmer — DISMISSED** [review]
only tested to a limited extend
if something does not work as expected - this needs to be fixed in a follow-up PR

the following packages were not yet ported to ros2 - i.e. they still have an `COLCON_IGNORE` file:
- `cob_cam3d_throttle`
- `cob_image_flip`
- `cob_object_detection_visualizer`
- `cob_perception_common`
- `cob_vision_utils`
- `ipa_3d_fov_visualization`

**Pull request merged**